central/networkgraph/aggregator/aggregator_impl.go

```diff
@@ -220,6 +221,12 @@ func updateDupNameExtSrcTracker(entity *storage.NetworkEntityInfo, dupNameExtSrc
 		return
 	}
 
+	if networkgraph.IsExternalDiscovered(entity) && !features.NetworkGraphExternalIPs.Enabled() {
+		// If this is a discovered entity, but network graph display is disabled
+		// we aggregate into the internet entity
+		*entity = *networkgraph.InternetEntity().ToProto()
```

Comment:
Why not simply
```suggestion
		entity = networkgraph.InternetEntity().ToProto()
```
?

Reply:
The goal is to anonymize (replace with "External source") the entity. With the proposed change, only the local copy of the pointer would be altered.

Reply:
I moved the mangling of discovered entities to a dedicated method `anonymizeIfNoExternalIPs()` to make it more explicit.

I made 2 different versions:

1.  using a reference to only mangle when necessary and avoid copies
2.  using a returned value to avoid using side-effects (as suggested by @vikin91 )

The PR currently has 2. on top.

Here is how 1. looks like:

https://github.com/stackrox/stackrox/blob/8ec2836d5cfbfc40b01d424b6ba78684a4a8a289/central/networkgraph/aggregator/aggregator_impl.go#L177-L181

Reply:
Actually, the linter forbids returning proto generated structs by value : https://github.com/stackrox/stackrox/actions/runs/11385817915/job/31676555459?pr=12901

So I am going to remove 2. as it is not viable.

For reference, 2. implementation :

https://github.com/stackrox/stackrox/blob/e6b193cf52183143bea5a512bb59a3ac8d52176d/central/networkgraph/aggregator/aggregator_impl.go#L177-L182

https://github.com/stackrox/stackrox/blob/e6b193cf52183143bea5a512bb59a3ac8d52176d/central/networkgraph/aggregator/aggregator_impl.go#L325-L337